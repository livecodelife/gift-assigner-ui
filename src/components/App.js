import axios from "axios";
import { useState } from "react";
import { AppWrapper, AssignGiftsButton, FamilyIdInput, GiftAssignForm } from "../styles/AppStyles";
import { GiftAssignments } from "./GiftAssignments";

function App() {
  const [assignments, setAssignments] = useState([])
  const [id, setId] = useState(null)
  const [error, setError] = useState(null)

  const getAssignments = async (e) => {
    setError(null)
    try {
      const assignmentData = await axios.get(`http://localhost:8080/assignments/${id}`)
      setAssignments(assignmentData.data)
    } catch (e) {
      setError('Something went wrong. Check your id and try again')
    }
  }

  return (
    <AppWrapper>
      <GiftAssignForm>
        <FamilyIdInput 
          type='number' 
          placeholder='ID' 
          max='3'
          onChange={(e) => setId(e.target.value)} 
        />
        <AssignGiftsButton onClick={getAssignments}>Get My Assignments!</AssignGiftsButton>
      </GiftAssignForm>
      {assignments.length > 0 && <GiftAssignments assignments={assignments} />}
      {error}
    </AppWrapper>
  );
}

export default App;
