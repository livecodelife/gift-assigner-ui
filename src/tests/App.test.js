import { shallow } from 'enzyme'
import App from '../components/App'

describe('Given an App component', () => {
  describe('When the component is rendered', () => {
    let wrapper = shallow(<App />)
    it('should have a gift assignment form', () => {
      expect(wrapper.find('GiftAssignForm').exists()).toBe(true)
    })
    it('should have a field for the familyId', () => {
      expect(wrapper.find('FamilyIdInput').exists()).toBe(true)
    })
    it('should have a button for assigning gifts', () => {
      expect(wrapper.find('AssignGiftsButton').exists()).toBe(true)
    })
  })
})
