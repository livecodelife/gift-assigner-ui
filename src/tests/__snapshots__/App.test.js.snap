// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When the component is rendered should match the snapshot 1`] = `
<styled.div>
  <GiftAssignForm>
    <FamilyIdInput
      onChange={[Function]}
      placeholder="ID"
      type="number"
    />
    <AssignGiftsButton
      onClick={[Function]}
    >
      Get My Assignments!
    </AssignGiftsButton>
  </GiftAssignForm>
</styled.div>
`;

exports[`Given an App component When the id is entered should show the gift assignments after clicking the button 1`] = `
<styled.div>
  <GiftAssignForm>
    <FamilyIdInput
      onChange={[Function]}
      placeholder="ID"
      type="number"
    />
    <AssignGiftsButton
      onClick={[Function]}
    >
      Get My Assignments!
    </AssignGiftsButton>
  </GiftAssignForm>
  <GiftAssignments
    assignments={
      Array [
        Object {
          "Gifter": "giftee",
        },
      ]
    }
  />
</styled.div>
`;
